tmt/schemas/results.yaml

```diff
@@ -135,6 +139,12 @@ items:
     duration:
       $ref: "#/definitions/duration"
 
+    # TODO: Fixme? e.g. using ref. otherwise I am getting the error:
```

Comment:
@happz Aren't we missing the `data-path` in the `result` and `subresult` schemas? Otherwise, I'm still getting the "result format violation" errors.

Reply:
Result schema is definitely missing `data-path` key, subresults shouldn't have one, IIUIC, data path is a path in which tmt stores test data and info and stuff, so there is no dedicated subpath for a subresult.

Reply:
Are you sure about the `data-path` in the subresult? I do not have much insight, but it seems the subresults generated by `tmt-report-result` are using this field when the `tmt-report-results.yaml` gets created:

- https://github.com/teemtee/tmt/blob/7b16db66907e9b2c62cc0643abd27a3a120e4a26/tmt/steps/execute/scripts/tmt-report-result#L20

Reply:
Hmmm, that does not look right. `TMT_TEST_EXE` is `${TMT_TREE/tree/execute}`, `TMT_TEST_DATA` is said to be "Path to the directory where test can store logs and other artifacts generated during its execution." and I can't imagine that `${TMT_TEST_DATA/$TMT_TREE/tree/execute}` could be anything sane.

@psss WDYT? There is "test data" for the test, `TMT_TEST_DATA` points to it, but data paths for Beakerlib phases? tmt does not manage those, in custom results we do not even have any idea whether the test used any subresult data paths, so we don't touch them, this seems like a mistake to me.

Reply:
Also, do you think we should somehow limit the `data-path` using a `pattern` in the schema?

For a parent result, the value is always a relative path (`data/guest/default-0/tests/example-1/data`). For subresults, the `data-path` seems to always start with `pattern: "^/"` - it's not relative, but absolute where root starts at 'execute' directory, e.g. `/data/guest/default-0/tests/example-1/data`.

Isn't it a bug? Also, the subresult's `data-path` seems to be always the same as the parent `data-path` (without `/` prefix).

Reply:
@psss What is your opinion about `data-path` for subresult? Thanks.

Reply:
I see the attribute is still part of the patch - @psss, please, chime in.

Reply:
@happz @psss Do you think we should remove `data-path` also from `tmt-report-results.yaml` generated by `tmt-report-result`?
- https://github.com/teemtee/tmt/blob/3803a4414365045a582ab13861580a0c25f1b642/tmt/steps/execute/scripts/tmt-report-result#L23

Or this one should be kept there for some reason (e.g. for `custom`/`restraint` results)?

Reply:
I'd probably try not emitting the key, there's no sane way to "guess" the data path anyway on this level, and tmt knows it and sets it for results when they are constructed (`data_path=invocation.relative_test_data_path` in `from_test_invocation()`).

Reply:
- [ ] AI: Ok. So I will create a separate PR for these changes which I think should be merged before this PR:
- Remove `data-path` for subresults (scheme and SubResult class)
- Remove `data-path` from `tmt-report-result` script & run the full test, see if something breaks + fix the tests

Reply:
I've pushed the changes, hopefully, this comment is now resolved. The removal of `data-path` from `tmt-report-results` will be covered in https://github.com/teemtee/tmt/pull/3270